### ISSUE TITLE
[Accessibility] Add keybindings going between terminal input and output buffer back and forth just like Copilot Chat view

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#189360]
Type: <b>Feature Request</b>

Currently, to navigate between terminal input and output, users are required to use the Tab and Shift+Tab keys. While this does offer a method for moving around, it's not an ideal solution. The tab cycle can include other tabbable elements which may not be relevant to the user's immediate needs.

I would like to point out the new navigation model introduced in the Copilot Chat view. This new navigation system is both efficient and user-friendly. By pressing Ctrl+UpArrow while in the chat input, users' focus is shifted directly to the chat output, regardless of how many times the Ctrl+UpArrow key is pressed. To return to the chat input from the output view, users simply press Ctrl+DownArrow.

By implementing a similar system for navigating between terminal input and output, VSCode could greatly improve the user experience. The intuitive and user-friendly design of the Copilot Chat view's navigation system makes it a perfect candidate to be extended into other areas of VSCode UI.

I believe many screen reader users would appreciate this change, as it would streamline their interactions with the terminal input and output buffer.

VS Code version: Code - Insiders 1.81.0-insider (9800cf6dd6bf4634889d60720ef46a400f3a7298, 2023-07-28T12:08:04.472Z)
OS version: Windows_NT x64 10.0.22621
Modes:

**Comment by jooyoungseo** [comment]
another benefit of having this navigation model is that we can address the Tab/Shift+Tab completion key conflict in terminal input area.

**Comment by meganrogge** [comment]
We'd need to do this first https://github.com/microsoft/vscode/issues/189358

**Comment by jooyoungseo** [comment]
@meganrogge We may want to discuss what keybindings to use for this. Ctrl+Up/DownArrow has been preoccupied by terminal scrolls. If we go with overriding Ctrl+Up/DownArrow keys, that would only work for Windows. We may want to avoid CMD+Up/DownArrow keybindings on Mac, as well. Also, we need to think about the key consistency between different platforms.

**Comment by jooyoungseo** [comment]
@meganrogge -- What I am concerned about is ...:

1. Alt+F2 is already used for different purpose: contextual help. We may want to keep the original behavior not to confuse users.

2. Besides muscle memory, modifier key+function key is not convenient to press. Unlike one-time-ish usage of contextual help, the frequent round trip between terminal input and buffer requires more easy-to-press keybindings. Plus, pressing function key on Mac touch bar for VoiceOver users require more time because you need to press FN key and double tap the function key in the touch bar.

**Comment by jooyoungseo** [comment]
@meganrogge I still think using modifier+Up/DownArrow keys are the best choices. In any event we cannot do this, we may consider IO model like below:

* Modifier key+I: this is for terminal input. Always guarantee to focus in the terminal input area when pressed in terminal buffer.

* Modifier key+O: This is output (terminal buffer) area. Always guarantees to focus in the terminal buffer area when pressed in terminal input.

IO model may be easy to remember and easy to press. Still, Mod+Up/DownArrows would be much easier to pick up.

**Comment by meganrogge** [comment]
> Alt+F2 is already used for different purpose: contextual help. We may want to keep the original behavior not to confuse users.

`Alt+F2` is used for the accessible view - which is basically what the accessible buffer is. Your other point makes sense though - thanks.

**Comment by jooyoungseo** [comment]
@meganrogge  Oh, my bad. yes, Alt+F1 is contextual help; Alt+F2 for Accessible View. My memory got messed up. I am sorry!

**Comment by Tyriar** [comment]
We could have ctrl/cmd+up automatically open the accessible view/buffer in the terminal, but ctrl/cmd+down should only be for navigation imo. Having ctrl/cmd+down always go back to the input means it would be difficult to navigate to a command higher if you need to backtrack (up n+1 times, down 1 time).

**Comment by meganrogge** [comment]
@Tyriar Cmd + up/down don't do anything atm in the terminal when in screen reader mode

**Comment by jooyoungseo** [comment]
@meganrogge -- Could we add a custom setting like typing any characters in terminal buffer moves focus to terminal input? Currently, typing in terminal buffer is just blocking the input and throwing a warning. Users can press Ctrl+UpArrow in the terminal input to focus on the buffer, and start typing to input.

**Comment by meganrogge** [comment]
To verify, enable screen reader mode. Focus the terminal. Use `ctrl/cmd+up/down arrows` to go between the terminal buffer and the terminal accessible buffer. Note that `ctrl/cmd+down arrow` should first go to the last line of the accessible buffer and then focus the terminal.
